book/fault_proofs/fault_proof_architecture.md

```diff
@@ -0,0 +1,148 @@
+# Architecture
+
+OP Succinct is also offered in "fault proof" mode, which is a good solution for chains that wish to have faster finality while not paying the ZK proving costs for every single block. The exact same OP Succinct ZK program is used for this mode--the only difference is the smart contrcts deployed onchain that integrate with OP Stack's `DisputeGameFactory` contract.
+
+## Overview
+
+We assume that the reader has a solid understanding of the OP Stack's `DisputeGameFactory` and `IDisputeGame` interface. Documentation can be found [here](https://specs.optimism.io/fault-proof/stage-one/dispute-game-interface.html). We implement the `IDisputeGame` interface with a ZK-enabled fault proof (using the OP Succinct SP1 program) instead of the standard interactive bisection game that the vanilla OP Stack uses.
+
+## Core Concepts
+
+1. **Proposals**: Each proposal represents a claimed state transition from a start L2 block to an end L2 block with a `startL2OutputRoot` and a `claimedL2OutputRoot` where the output root is a commitment to the entirety of L2 state.
+2. **Challenges**: Participants can challenge proposals they believe are invalid.
+3. **Proofs**: ZK proofs that verify the correctness of state transitions contained in proposals, anchored against an L1 block hash.
+4. **Resolution**: Process of determining whether a proposal is valid or not.
+
+## Dispute Game Implementation
+
+Proposing new state roots goes through the regular flow of the `DisputeGameFactory` to the `OPSuccinctFaultDisputeGame` contract that implements the `IDisputeGame` interface. Each proposal contains a link to a previous parent proposal (unless it is the first proposal after initialization, in which case it stores the previous parent as `addr(0)`), and includes a `l2BlockNumber` and claimed `l2OutputRoot`.
```

Comment:
```suggestion
Proposing new state roots goes through the regular flow of the `DisputeGameFactory` to the `OPSuccinctFaultDisputeGame` contract that implements the `IDisputeGame` interface. Each proposal contains a link to a previous parent proposal and includes a `l2BlockNumber` and claimed `l2OutputRoot`.
```
For a new game type, anchor state registry is initialized with a starting anchor root.